File: 14/14.py
def collatz(num):
  length = 1
  while num > 1:
    num = num/2 if num % 2 == 0 else 3 * num + 1
    length += 1
  return length

longest = (0,0)
for i in xrange(1000000):
  length = collatz(i)
  longest = (i, length) if length > longest[1] else longest
print longest[0]

File: 14/14.c
#include <stdio.h>

int collatz(long n) {
  int i;
  for (i = 0; n != 1; i++) {
    n = (n && !(n % 2)) ? n / 2 : (n * 3) + 1;
  }
  return i;
}

int main() {
  int highest[2] = {0,0};
  for (int i = 1; i < 1000000; i += 2) {
    int x = collatz(i);
    if (x > highest[1]) {
      highest[0] = i;
      highest[1] = x;
    }
  }
  printf("%d\n", highest[0]);
}
